lms/djangoapps/course_goals/models.py

```diff
@@ -33,6 +34,13 @@ class Meta:
     user = models.ForeignKey(User, blank=False, on_delete=models.CASCADE)
     course_key = CourseKeyField(max_length=255, db_index=True)
     goal_key = models.CharField(max_length=100, choices=GOAL_KEY_CHOICES, default=GOAL_KEY_CHOICES.unsure)
+    # null=True is for original creation where we don't know how those rows were created.
+    # All future rows should specify a value.
```

Comment:
I have an open question in slack about if there is a way to avoid this where we can still set to null, but disallow it from the future. In case that doesn't come up fruitful, I believe this is acceptable.

Reply:
Seems like not. We decided to override the save method to shortcircuit if a non Boolean value is provided.